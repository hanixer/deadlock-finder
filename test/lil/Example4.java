package showcase;

/**
 * Nested loop with break and continue
 */
public class Example4 {
    static void func1() {
        int x = 6;
        int y = 3;
        while (x < 100) {
            x = x + 1;
            if (y == 50) 
                break;
            while (x > 4) {
                if (y == 23) 
                    continue;
                x = x - 1;
            }
        }
    }
}