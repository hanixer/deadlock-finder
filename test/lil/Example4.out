func showcase.Example4.func1(): void
entry:
jump bb1

bb1:
var x: int = 6
var y: int = 3
jump bb2

bb2:
var t~1: boolean = x<100
condJump t~1 bb3 end

bb3:
x = x+1
var t~2: boolean = y==50
condJump t~2 bb5 bb4

bb5:
jump end

bb4:
var t~3: boolean = x>4
condJump t~3 bb6 bb2

bb6:
var t~4: boolean = y==23
condJump t~4 bb8 bb7

bb8:
jump bb4

bb7:
x = x-1
jump bb4

end:
return